integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+                    .innerJoin(otherBook).on(otherBook.owner.eq(otherAuthor))
+                    .transform(GroupBy.groupBy(otherAuthor).as(GroupBy.list(otherBook)));
+
+            assertNotNull(booksByAuthor);
+        });
+    }
+
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class })
+    public void testFromValues() {
+        doInJPA(entityManager -> {
+            Document theBook = new Document();
+            theBook.setId(1337L);
+            theBook.setName("test");
+
+            List<Document> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .fromValues(document, Collections.singleton(theBook))
+                    .select(document)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class })
+    public void testFromValuesAttributes() {
+        doInJPA(entityManager -> {
+            StringPath bookName = Expressions.stringPath("bookName");
+
+            List<String> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .fromValues(document.name, bookName, Collections.singleton("book"))
+                    .select(bookName)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoMySQL.class, NoFirebird.class, NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class })
+    public void  testComplexUnion() {
+        doInJPA(entityManager -> {
+            Person person = new Person();
+            person.setName("Person");
+            entityManager.persist(person);
+
+            Document theBook = new Document();
+            theBook.setId(1337L);
+            theBook.setName("test");
+            theBook.setOwner(person);
+            entityManager.merge(theBook);
+
+            Document theSequel = new Document();
+            theSequel.setId(42L);
+            theSequel.setName("test2");
+            theSequel.setOwner(person);
+            entityManager.merge(theSequel);
+        });
+
+        doInJPA(entityManager -> {
+            List<Document> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .union(
+                        new BlazeJPAQuery<Document>().intersect(
+                                select(document).from(document).where(document.id.eq(41L)),
+                                new BlazeJPAQuery<Document>().except(
+                                        select(document).from(document).where(document.id.eq(42L)),
+                                        select(document).from(document).where(document.id.eq(43L)))),
+                        select(document).from(document).where(document.id.eq(46L)))
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoMySQL.class, NoFirebird.class, NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class })
```

Comment:
Add a comment like e.g. `// NOTE: No advanced sql support for datanucleus, eclipselink and openjpa yet` and also `// NOTE: MySQL only supports the UNION set operation`